src/sagas/backend.ts

```diff
@@ -277,6 +278,67 @@ function* backendSaga(): SagaIterator {
       handleResponseError(resp);
     }
   });
+
+  yield takeEvery(actionTypes.FETCH_NOTIFICATIONS, function*(action) {
+    const tokens = yield select((state: IState) => ({
+      accessToken: state.session.accessToken,
+      refreshToken: state.session.refreshToken
+    }));
+
+    if (!tokens.accessToken) {
+      return;
+    }
+
+    const notifications = yield getNotifications(tokens);
+
+    yield put(actions.updateNotifications(notifications));
+  });
+
+  yield takeEvery(actionTypes.ACKNOWLEDGE_NOTIFICATION, function*(action) {
+    const tokens = yield select((state: IState) => ({
+      accessToken: state.session.accessToken,
+      refreshToken: state.session.refreshToken
+    }));
+    const ids = (action as actionTypes.IAction).payload as number[];
+
+    if (ids.length === 0) {
+      return;
+    }
+
+    const notifications: Notification[] = yield select(
+      (state: IState) => state.session.notifications
+    );
+    const newNotifications: Notification[] = notifications.filter(
+      notification => !ids.includes(notification.id)
+    );
+
+    /* 
+      For responsiveness, we update the store's state first, then
+      check if the server rejects the acknowledgement and undo the changes if necessary
+    */
+
```

Comment:
Here, I don't see the need to undo the changes. If the acknowledgement fails, when the notifications are fetched again from the backend, it will be there?

Furthermore, how slow is it really? Is it a noticeable difference?

Reply:
I think it could become potentially noticeable. I guess I should revert it such that the notifications are updated on successful response?

Reply:
I just feel its messy to update and rollback when the response fails. I prefer either updating immediately (and let the notifications load again on fetch notification if updating fails) or updating on success.

Reply:
Ok, then it will update immediately